### PR TITLE
init module

### PR DESCRIPTION
**Pull request by filippolmt**

### CODE REVIEW

File: values/kube-prometheus-stack.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    requests:
+      cpu: 50m
+      memory: 100Mi
+
+prometheus:
+  prometheusSpec:
+    resources:
+      requests:
+        cpu: 1300m
+        memory: 1500Mi
```
Comment (paolomainardi) [inline values/kube-prometheus-stack.yml:74]: make them variables

File: values/adapter-values.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+rules:
+  default: true
+  custom:
+  existing:
+  external: []
+
+resources:
+  requests:
+    cpu: 1000m
+    memory: 384Mi
```
Comment (paolomainardi) [inline values/adapter-values.yml:23]: variables

**Comment by paolomainardi** [comment]
Add a README

File: values/kube-prometheus-stack.yml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        cpu: ${prometheus_resources.cpu_requests}
+        memory: ${prometheus_resources.memory_requests}
+  %{~ endif ~}  
+    storageSpec:
+      volumeClaimTemplate:
+        spec:
+        %{~ if trimspace(storage_class_name) != "" ~}
+          storageClassName: ${storage_class_name}
+        %{~ endif ~}
+          accessModes: [ "ReadWriteOnce" ]
```
Comment (Monska85) [inline values/kube-prometheus-stack.yml:82]: If this value is the default we can omit it

File: values/kube-prometheus-stack.yml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+kubeEtcd:
+  enabled: false
+
+# Disable Controller metrics
+kubeControllerManager:
+  enabled: false
+
+# Disable Scheduler metrics
+kubeScheduler:
+  enabled: false
```
Comment (Monska85) [inline values/kube-prometheus-stack.yml:97]: I think that these configurations should be a variables

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  type = string
+}
+
+variable "storage_class_name" {
+  type = string
+  default = ""
+}
+
+variable "prometheus_resources" {
+  type        = any
```
Comment (Monska85) [inline variables.tf:66]: I thinkt that here we can define a specific object, like:

```suggestion
  type        = object({ cpu_requests = "",  memory_requests ="" })
```

What do you think?

File: variables.tf [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+  description = "Prometheus resource"
+  default     = { 
+                  cpu_requests = "500m", 
+                  memory_requests = "512Mi" 
+                }
+}
```
Comment (Monska85) [inline variables.tf:72]: If I remember correctly we decided to don't provide default values for the resources configuration

File: variables.tf [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+  default     = { 
+                  cpu_requests = "50m", 
+                  memory_requests = "100Mi" 
+                }
+}
+
+variable "grafana_resources" {
+  type        = any
+  description = "Grafana resource"
+  default     = { cpu_requests = "100",
```
Comment (Monska85) [inline variables.tf:86]: Be consistent in the data structure indentation

File: values/kube-prometheus-stack.yml (hunk 4 above)
Comment (filippolmt) [reply values/kube-prometheus-stack.yml]: Done

File: variables.tf (hunk 5 above)
Comment (filippolmt) [reply variables.tf:67]: I added this:
```
default     = { 
                  cpu_requests = "", 
                  memory_requests = "" 
                }
```
what do you think?

File: variables.tf (hunk 6 above)
Comment (filippolmt) [reply variables.tf]: done

File: variables.tf (hunk 7 above)
Comment (filippolmt) [reply variables.tf]: done

File: values/kube-prometheus-stack.yml [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    ##
+    inMemory:
+      enabled: false
+      ## The maximum usage on memory medium EmptyDir would be
+      ## the minimum value between the SizeLimit specified
+      ## here and the sum of memory limits of all containers in a pod
+      ##
+      # sizeLimit: 300Mi
+prometheusOperator:
+  %{~ if prometheus_operator_resources != "" ~}
```
[the remark this answers is not in the document]
Comment (filippolmt) [reply values/kube-prometheus-stack.yml]: done

File: values/kube-prometheus-stack.yml [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+    ## If persistence is not enabled, this allows to mount the
+    ## local storage in-memory to improve performance
+    ##
+    inMemory:
+      enabled: false
+      ## The maximum usage on memory medium EmptyDir would be
+      ## the minimum value between the SizeLimit specified
+      ## here and the sum of memory limits of all containers in a pod
+      ##
+      # sizeLimit: 300Mi
```
[the remark this answers is not in the document]
Comment (filippolmt) [reply values/kube-prometheus-stack.yml]: done

File: values/kube-prometheus-stack.yml [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+##
+grafana:
+  ingress:
+    ## If true, Grafana Ingress will be created
+    ##
+    enabled: true
+    annotations:
+      kubernetes.io/ingress.class: nginx
+      cert-manager.io/cluster-issuer: "${cluster_issuer_name}"
+      acme.cert-manager.io/http01-edit-in-place: "true"
```
[the remark this answers is not in the document]
Comment (filippolmt) [reply values/kube-prometheus-stack.yml:24]: Fixed, do you think the solution I have adopted would be okay?

File: values/kube-prometheus-stack.yml [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+  %{~ endif ~}
+## Using default values from https://github.com/grafana/helm-charts/blob/main/charts/grafana/values.yaml
+##
+grafana:
+  ingress:
+    ## If true, Grafana Ingress will be created
+    ##
+    enabled: true
+    annotations:
+      kubernetes.io/ingress.class: nginx
```
[the remark this answers is not in the document]
Comment (filippolmt) [reply values/kube-prometheus-stack.yml]: done

File: main.tf [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+resource "random_password" "grafana_admin_password" {
+  length           = 30
+  special          = true
+  override_special = "_%@"
+}
+
+resource "kubernetes_namespace" "kube_prometheus_stack_namespace" {
+  count = local.needs_auth ? 1 : 0
```
[the remark this answers is not in the document]
Comment (filippolmt) [reply main.tf]: done

File: values/kube-prometheus-stack.yml (hunk 1 above)
Comment (filippolmt) [reply values/kube-prometheus-stack.yml]: done

File: values/kube-prometheus-stack.yml (hunk 10 above)
[the remark this answers is not in the document]
Comment (Monska85) [reply values/kube-prometheus-stack.yml:24]: LGTM!

File: variables.tf (hunk 5 above)
Comment (Monska85) [reply variables.tf:67]: This is ok for the default value, but we can also force the variable to be an object with these two fields.

**Pull request merged**